week3/homework/src/actions/readTodo.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+async function readTodo(todo, req, res) {
+  try {
+    const todoList = await todo.readList().then(todoList => JSON.parse(todoList));
+    const index = parseInt(req.params.id);
+    if (index > 0 && index <= todoList.length) {
+      res.status(200).send(todoList[index - 1]);
+    } else {
+      res.status(400).send({ Failed: 'Please provide a valid id ' });

```

Comment:
this seems to be repeated also above. This means it needs to be put in a separate function. Can you think of a way to achieve this?